akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpMessageParser.scala

```diff
@@ -178,8 +178,19 @@ private[http] trait HttpMessageParser[Output >: MessageOutput <: ParserOutput] {
             }
         }
         case h: `Transfer-Encoding` => teh match {
-          case None    => parseHeaderLines(input, lineEnd, headers, headerCount + 1, ch, clh, cth, Some(h), e100c, hh)
-          case Some(x) => parseHeaderLines(input, lineEnd, headers, headerCount, ch, clh, cth, Some(x append h.encodings), e100c, hh)
+          case None =>
+            h.encodings match {
+              case Seq(TransferEncodings.chunked) =>
+                // A single chunked is the only one we support
+                parseHeaderLines(input, lineEnd, headers, headerCount + 1, ch, clh, cth, Some(h), e100c, hh)
```

Comment:
We could change the interface here to avoid any ambiguities and make that flag a `isChunked: Boolean`. In any case, the `Content-Length` vs. `Transfer-Encoding` check in `parseEntity` needs to be changed, so that the check will work for any Transfer-Encoding header passed to the method.

Reply:
Went with changing it into a boolean since that is more explicit about what is actually supported.

One curious thing now is that we support rendering with whatever Transfer-Encoding but not parsing, but maybe that is to be expected?

Reply:
Added https://github.com/akka/akka-http/issues/3761 to follow up on that.